Installing `ndg-httpsclient==0.3.3` resolved the ssl issue. Is this something that could be added to the `install_requires` (preferably only for python 2)? Here are some logs showing how it worked (second failure is due to not having key / secret set).

```
(mushu-sni)[vagrant@vagrant mushu]$ inv halo.list_groups
Traceback (most recent call last):
  File "/data/virtualenv/mushu-sni/bin/inv", line 9, in <module>
    load_entry_point('invoke==0.9.0', 'console_scripts', 'inv')()
  File "/data/virtualenv/mushu-sni/lib/python2.7/site-packages/invoke/cli.py", line 304, in main
    dispatch(sys.argv)
  File "/data/virtualenv/mushu-sni/lib/python2.7/site-packages/invoke/cli.py", line 297, in dispatch
    return executor.execute(*tasks, dedupe=dedupe)
  File "/data/virtualenv/mushu-sni/lib/python2.7/site-packages/invoke/executor.py", line 90, in execute
    task=task, name=name, args=args, kwargs=kwargs
  File "/data/virtualenv/mushu-sni/lib/python2.7/site-packages/invoke/executor.py", line 131, in _execute
    result = task(*args, **kwargs)
  File "/data/virtualenv/mushu-sni/lib/python2.7/site-packages/invoke/tasks.py", line 111, in __call__
    result = self.body(*args, **kwargs)
  File "/vagrant/src/sandbox/mushu/tooling/halo.py", line 23, in list_groups
    cp = CloudPassage(key_id, secret_key)
  File "/data/virtualenv/mushu-sni/lib/python2.7/site-packages/cloudpassage/cloudpassage.py", line 86, in __init__
    self.token.login(keyid, secret_key)
  File "/data/virtualenv/mushu-sni/lib/python2.7/site-packages/cloudpassage/cloudpassage.py", line 46, in login
    self.token()
  File "/data/virtualenv/mushu-sni/lib/python2.7/site-packages/cloudpassage/cloudpassage.py", line 55, in token
    json = self.__post(endpoint, None, headers)
  File "/data/virtualenv/mushu-sni/lib/python2.7/site-packages/cloudpassage/cloudpassage.py", line 26, in __post
    req = requests.post(url, data=data, headers=headers)
  File "/data/virtualenv/mushu-sni/lib/python2.7/site-packages/requests/api.py", line 94, in post
    return request('post', url, data=data, json=json, **kwargs)
  File "/data/virtualenv/mushu-sni/lib/python2.7/site-packages/requests/api.py", line 49, in request
    return session.request(method=method, url=url, **kwargs)
  File "/data/virtualenv/mushu-sni/lib/python2.7/site-packages/requests/sessions.py", line 457, in request
    resp = self.send(prep, **send_kwargs)
  File "/data/virtualenv/mushu-sni/lib/python2.7/site-packages/requests/sessions.py", line 569, in send
    r = adapter.send(request, **kwargs)
  File "/data/virtualenv/mushu-sni/lib/python2.7/site-packages/requests/adapters.py", line 420, in send
    raise SSLError(e, request=request)
requests.exceptions.SSLError: hostname 'api.cloudpassage.com' doesn't match 'grid.cloudpassage.com'
(mushu-sni)[vagrant@vagrant mushu]$ pip install pyOpenSSL
Requirement already satisfied (use --upgrade to upgrade): pyOpenSSL in /data/virtualenv/mushu-sni/lib/python2.7/site-packages
Requirement already satisfied (use --upgrade to upgrade): cryptography>=0.2.1 in /data/virtualenv/mushu-sni/lib/python2.7/site-packages (from pyOpenSSL)
Requirement already satisfied (use --upgrade to upgrade): six>=1.5.2 in /data/virtualenv/mushu-sni/lib/python2.7/site-packages (from pyOpenSSL)
Requirement already satisfied (use --upgrade to upgrade): cffi>=0.8 in /data/virtualenv/mushu-sni/lib/python2.7/site-packages (from cryptography>=0.2.1->pyOpenSSL)
Requirement already satisfied (use --upgrade to upgrade): setuptools in /data/virtualenv/mushu-sni/lib/python2.7/site-packages (from cryptography>=0.2.1->pyOpenSSL)
Requirement already satisfied (use --upgrade to upgrade): pycparser in /data/virtualenv/mushu-sni/lib/python2.7/site-packages (from cffi>=0.8->cryptography>=0.2.1->pyOpenSSL)
Cleaning up...
(mushu-sni)[vagrant@vagrant mushu]$ pip install ndg-httpsclient
Downloading/unpacking ndg-httpsclient
  Downloading ndg_httpsclient-0.3.3.tar.gz
  Storing download in cache at /tmp/pip/https%3A%2F%2Flocalshop.waveapps.com%2Fpackages%2Fndg-httpsclient%2Fdownload%2F14526%2Fndg_httpsclient-0.3.3.tar.gz
  Running setup.py (path:/data/virtualenv/mushu-sni/build/ndg-httpsclient/setup.py) egg_info for package ndg-httpsclient

Requirement already satisfied (use --upgrade to upgrade): PyOpenSSL in /data/virtualenv/mushu-sni/lib/python2.7/site-packages (from ndg-httpsclient)
Requirement already satisfied (use --upgrade to upgrade): cryptography>=0.2.1 in /data/virtualenv/mushu-sni/lib/python2.7/site-packages (from PyOpenSSL->ndg-httpsclient)
Requirement already satisfied (use --upgrade to upgrade): six>=1.5.2 in /data/virtualenv/mushu-sni/lib/python2.7/site-packages (from PyOpenSSL->ndg-httpsclient)
Requirement already satisfied (use --upgrade to upgrade): cffi>=0.8 in /data/virtualenv/mushu-sni/lib/python2.7/site-packages (from cryptography>=0.2.1->PyOpenSSL->ndg-httpsclient)
Requirement already satisfied (use --upgrade to upgrade): setuptools in /data/virtualenv/mushu-sni/lib/python2.7/site-packages (from cryptography>=0.2.1->PyOpenSSL->ndg-httpsclient)
Requirement already satisfied (use --upgrade to upgrade): pycparser in /data/virtualenv/mushu-sni/lib/python2.7/site-packages (from cffi>=0.8->cryptography>=0.2.1->PyOpenSSL->ndg-httpsclient)
Installing collected packages: ndg-httpsclient
  Running setup.py install for ndg-httpsclient
    Skipping installation of /data/virtualenv/mushu-sni/lib/python2.7/site-packages/ndg/__init__.py (namespace package)

    Installing /data/virtualenv/mushu-sni/lib/python2.7/site-packages/ndg_httpsclient-0.3.3-py2.7-nspkg.pth
    Installing ndg_httpclient script to /data/virtualenv/mushu-sni/bin
Successfully installed ndg-httpsclient
Cleaning up...
(mushu-sni)[vagrant@vagrant mushu]$ pip install pyasn1
Requirement already satisfied (use --upgrade to upgrade): pyasn1 in /data/virtualenv/mushu-sni/lib/python2.7/site-packages
Cleaning up...
(mushu-sni)[vagrant@vagrant mushu]$ inv halo.list_groups
Traceback (most recent call last):
  File "/data/virtualenv/mushu-sni/bin/inv", line 9, in <module>
    load_entry_point('invoke==0.9.0', 'console_scripts', 'inv')()
  File "/data/virtualenv/mushu-sni/lib/python2.7/site-packages/invoke/cli.py", line 304, in main
    dispatch(sys.argv)
  File "/data/virtualenv/mushu-sni/lib/python2.7/site-packages/invoke/cli.py", line 297, in dispatch
    return executor.execute(*tasks, dedupe=dedupe)
  File "/data/virtualenv/mushu-sni/lib/python2.7/site-packages/invoke/executor.py", line 90, in execute
    task=task, name=name, args=args, kwargs=kwargs
  File "/data/virtualenv/mushu-sni/lib/python2.7/site-packages/invoke/executor.py", line 131, in _execute
    result = task(*args, **kwargs)
  File "/data/virtualenv/mushu-sni/lib/python2.7/site-packages/invoke/tasks.py", line 111, in __call__
    result = self.body(*args, **kwargs)
  File "/vagrant/src/sandbox/mushu/tooling/halo.py", line 23, in list_groups
    cp = CloudPassage(key_id, secret_key)
  File "/data/virtualenv/mushu-sni/lib/python2.7/site-packages/cloudpassage/cloudpassage.py", line 86, in __init__
    self.token.login(keyid, secret_key)
  File "/data/virtualenv/mushu-sni/lib/python2.7/site-packages/cloudpassage/cloudpassage.py", line 46, in login
    self.token()
  File "/data/virtualenv/mushu-sni/lib/python2.7/site-packages/cloudpassage/cloudpassage.py", line 55, in token
    json = self.__post(endpoint, None, headers)
  File "/data/virtualenv/mushu-sni/lib/python2.7/site-packages/cloudpassage/cloudpassage.py", line 27, in __post
    req.raise_for_status()
  File "/data/virtualenv/mushu-sni/lib/python2.7/site-packages/requests/models.py", line 825, in raise_for_status
    raise HTTPError(http_error_msg, response=self)
requests.exceptions.HTTPError: 401 Client Error: Unauthorized
```

--- a/setup.py
+++ b/setup.py
@@ -1,5 +1,6 @@
 #!/usr/bin/env python
 import os
+import sys
 try:
     from setuptools import setup
 except ImportError:
@@ -9,6 +10,15 @@ def read(*paths):
     """Build a file path from *paths* and return the contents."""
     with open(os.path.join(*paths), 'r') as f:
         return f.read()
+
+install_requires = ['requests >= 2.1.0']
+
+# For SNI support (required by CloudPassage) in Python 2, must install the
+# following packages
+if sys.version_info[0] == 2:
+    install_requires.append('pyOpenSSL >= 0.14')
+    install_requires.append('ndg-httpsclient >= 0.3.3')
+    install_requires.append('pyasn1 >= 0.1.7')
 
 setup(
     name='cloudpassage',
@@ -23,7 +33,7 @@ setup(
     author='Ryan Gooler',
     author_email='ryan.gooler@gmail.com',
     py_modules=['cloudpassage'],
-    install_requires=['requests >= 2.1.0'],
+    install_requires=install_requires,
     include_package_data=True,
     classifiers=[
         'Development Status :: 3 - Alpha',

